context/app/static/js/shared-styles/tables/EntitiesTable/EntityTable.tsx

```diff
@@ -0,0 +1,157 @@
+  isSelectable: boolean;
+}
+
+const headerRowHeight = 60;
+
+function EntityTable<Doc>({ query, columns, isSelectable = true }: EntityTableProps<Doc>) {
+  const columnNameMapping = columns.reduce((acc, column) => ({ ...acc, [column.id]: column.sort }), {});
+
+  const {
+    searchHits,
+    allSearchIDs,
+    isLoading,
+    sortState,
+    setSort,
+    fetchMoreOnBottomReached,
+    virtualRows,
+    tableBodyPadding,
+    tableContainerRef,
+  } = useScrollTable<Doc>({
+    query,
+    columnNameMapping,
+    initialSortState: { columnId: 'last_modified_timestamp', direction: 'desc' },
+  });
+
+  return (
+    <StyledTableContainer
+      component={Paper}
+      ref={tableContainerRef}
+      onScroll={(event) => fetchMoreOnBottomReached(event)}
+    >
+      <Table stickyHeader>
+        <TableHead sx={{ position: 'relative' }}>
+          <TableRow sx={{ height: headerRowHeight }}>
+            {isSelectable && (
+              <SelectableHeaderCell
+                allTableRowKeys={allSearchIDs}
+                disabled={false}
+                sx={({ palette }) => ({ backgroundColor: palette.background.paper })}
+              />
+            )}
+            {columns.map((column) => (
+              <EntityHeaderCell column={column} setSort={setSort} sortState={sortState} key={column.id} />
+            ))}
+          </TableRow>
+          <TableRow aria-hidden="true">
+            <TableCell
+              colSpan={columns.length + 1}
+              sx={{ top: headerRowHeight, border: 'none' }}
+              padding="none"
+              component="td"
+            >
+              <LinearProgress
+                sx={{
+                  width: '100%',
+                  opacity: isLoading ? 1 : 0,
+                  transition: 'opacity',
+                }}
+                variant="indeterminate"
+              />
+            </TableCell>
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {tableBodyPadding.top > 0 && (
+            <tr>
+              <td style={{ height: `${tableBodyPadding.top}px` }} aria-hidden="true" />
+            </tr>
+          )}
+          {virtualRows.map((virtualRow) => {
+            const hit = searchHits[virtualRow.index];
+            if (hit) {
+              return (
+                <TableRow sx={{ height: virtualRow.size }} key={hit?._id}>
+                  {isSelectable && <SelectableRowCell rowKey={hit?._id} />}
+                  {columns.map(({ cellContent: CellContent, id }) => (
+                    <TableCell key={id}>
+                      <CellContent hit={hit._source} />
+                    </TableCell>
+                  ))}
+                </TableRow>
+              );
+            }
+            return null;
+          })}
+          {tableBodyPadding.bottom > 0 && (
+            <tr>
+              <td style={{ height: `${tableBodyPadding.bottom}px` }} aria-hidden="true" />
```

Comment:
Given this block has a lot in common with the block on lines 125-129, should we consider adding a `TablePadding` component to create an empty row like this?

```jsx
function TablePadding({ height }: { height?: number }) {
  if (!height || height <== 0) {
    return null;
  } 
  return (
    <tr aria-hidden="true">
      <Box component="td" height={height} />
    </tr>
  )
}
```